### PR TITLE
Code cleanup. Make the script shellcheck compliant. Including removin…

### PR DESCRIPTION
**Pull request by sigboe**
Code cleanup. Make the script shellcheck compliant. Including removing some edge case bugs.

Still falsely trips tests for [SC2059](https://github.com/koalaman/shellcheck/wiki/SC2059) and [SC2094](https://github.com/koalaman/shellcheck/wiki/SC2094) but I did not put in an ignore for them. I can add ignores for these tests if anyone wishes.

Also breaks out test for if aria2c is installed to a variable, and then uses the preferred downloader via `eval` to remove possibility of wrongly reading the exit status to check if downloading failed.

added a function to exit with an error message. Notably used if any `cd` command fails.

### CODE REVIEW

File: pineapple.sh [hunk 1]
```diff
@@ -75,43 +79,39 @@ elif [ "$option" = "4" ]; then
...
-if [ $? -ne 0 ]; then
-    printf "Download failed!\n"
-    printf "If you are in Italy or Iran, please use a VPN in another country\n"
-    printf "otherwise, please try again in a few minutes\n"
-    exit
+    eval "$downloader" "$(grep -o 'https://cdn-.*.7z' version.txt | head -n 1)" || printf '%s\n' \
+"Download failed!
+If you are in Italy or Iran, please use a VPN in another country
+otherwise, please try again in a few minutes"
+
```
Comment (MGThePro) [inline pineapple.sh:99]: removing the exit here just makes the script continue on a failed download. Since we leave the 7z files in the folder we're working in, this could lead to the script installing some wrong version that was already in /tmp/pineapple

File: pineapple.sh [hunk 2]
```diff
@@ -39,25 +43,25 @@ prompt()
...
+	title="$latest"
+	curl -s "$(grep -m1 -o 'https.*7z' version.txt )" > version.txt
 elif [ "$option" = "2" ]; then
 	printf "Available versions:\n"
 	uniq version.txt | grep -o 'EA .*' | tr -d '</a><br>' | sed -e ':a;N;$!ba;s/\n/,/g' -e 's/\EA //g'
 	printf "Choose version number:"
-	read version <&1
-	title=$version
-	curl -s $(grep "YuzuEA-$version" version.txt | grep -o 'https.*7z') > version.txt
+	read -r version <&1
```
Comment (MGThePro) [inline pineapple.sh:62]: For some reason the added -r breaks the version selector for me

File: pineapple.sh (hunk 1 above)
Comment (sigboe) [reply pineapple.sh:99]: It was a mistake to remove the exit there. Ill fix that.

File: pineapple.sh (hunk 2 above)
Comment (sigboe) [reply pineapple.sh:62]: Thanks for finding this, I'll figure out why it breaks and apply another safe workaround or revert that change.

**Comment by MGThePro** [comment]
Closing for now. Feel free to open another pull request once you update this and fix any remaining issues.
